src/components/GlobalNav/GlobalNav.tsx

```diff
@@ -0,0 +1,314 @@
+import {
+  Button,
+  Flex,
+  Icon,
+  Text,
+  View,
+  VisuallyHidden,
+} from "@aws-amplify/ui-react";
+import {
+  createContext,
+  useMemo,
+  useState,
+  useLayoutEffect,
+  useRef,
+  useEffect,
+} from "react";
+import styles from "./GlobalNav.module.scss";
+import { NavMenuLink } from "./components/NavMenuLink";
+import { ChevronDownIcon, ChevronUpIcon } from "./components/icons";
+
+export interface NavMenuItem {
+  type: "DEFAULT" | "EXTERNAL" | "ICON";
+  label: string;
+  url: string;
+  order: number;
+  placement: "LEFT" | "RIGHT";
+  icon?: string;
+}
+
+interface NavProps {
+  links: NavMenuItem[];
+  currentSite: string;
+  secondaryNavDesktop?: JSX.Element;
+  secondaryNavMobile?: JSX.Element;
+  setIsMobileState?: any;
+}
+
+type NavMobileContextType = {
+  showGlobalNav: boolean;
+  setShowGlobalNav: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const NavMobileContext = createContext<NavMobileContextType>({
+  showGlobalNav: false,
+  setShowGlobalNav: () => ({}),
+});
+
+export function GlobalNav({
+  links,
+  currentSite,
+  secondaryNavDesktop,
+  secondaryNavMobile,
+}: NavProps) {
+  let hasSecondaryNav = false;
+
+  if (secondaryNavDesktop && secondaryNavMobile) {
+    hasSecondaryNav = true;
+  }
+
+  const leftLinks: NavMenuItem[] = links.filter((e) => e.placement === "LEFT");
+  const rightLinks: NavMenuItem[] = links.filter(
+    (e) => e.placement === "RIGHT"
+  );
+
+  leftLinks.sort((a, b) => a.order - b.order);
+  rightLinks.sort((a, b) => a.order - b.order);
+
+  const [isMobileState, setIsMobileState] = useState(false);
+  const [mobileNavBreakpoint, setMobileNavBreakpoint] = useState(0);
+  const [currentWindowInnerWidth, setCurrentWindowInnerWidth] = useState(
+    window.innerWidth
+  );
+
+  const navLinksContainerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
```

Comment:
Thanks Jacob! I originally was using them, but there were instances where the text would wrap inside the nav bar. I can add more comments to the code, but basically I had to add `white-space: nowrap` to each of the nav bar links and then see if it causes it to scroll so that I could just show the mobile view instead

I can try it out with your example though and let you know how it goes!